core/src/main/java/org/ehcache/config/ResourcePoolsBuilder.java

```diff
@@ -38,6 +38,19 @@ public static ResourcePoolsBuilder newResourcePoolsBuilder() {
     return new ResourcePoolsBuilder();
   }
 
+  public static ResourcePoolsBuilder newResourcePoolsBuilder(ResourcePools pools) {
+    ResourcePoolsBuilder poolsBuilder = new ResourcePoolsBuilder();
+    for (ResourceType currentResourceType : pools.getResourceTypeSet()) {
+      Boolean persistence = false;
+      if(currentResourceType.isPersistable()) {
```

Comment:
Why the check? Trust that the pools you get is valid.

If it was not, the proper handling would be to throw as otherwise you actually change the pools definition you are copying.